NMS-14182: more alarmd doc tweaks (#4648)
OpenNMS/opennms: docs/modules/operation/pages/alarms/configuring-alarms.adoc

    <alarm-data reduction-key="%uei%:%dpname%:%nodeid%" alarm-type="1" auto-clean="false"/>
</event>
----
The `$dpname%` portion of the key refers to the "distributed poller name", which would be the name of the Minion that originated the event.
====

.Least Significant reduction-key Attribute
====

> Decreasing the significance of the reduction-key is a way to aggregate, for example, all nodes down in to a single alarm.

Reviewer: ```suggestion
Decreasing the significance of the reduction-key is a way to aggregate, for example, all nodes into a single alarm.
```

====

== Alarm-type attribute
Alarmd is designed to automatically match resolving events with an existing alarm.
Alarms with matching resolutions with problems (Ups with Downs), should be indicated with the alarm-type attribute.
There are currently three types of alarms:

* alarm-type="1" (problem alarm)
* alarm-type="2" (resolving alarm)

> * alarm-type="3" (notification alarm... alarm with no resolution such as SNMP Authentication Failures)

Reviewer: ```suggestion
* alarm-type="3" (notification alarm ... alarm with no resolution, such as SNMP authentication failures)
```

This attribute is used in the pairwise correlation feature of Alarmd.
When configuring a resolution alarm, set this attribute to match the reduction-key of the corresponding problem alarm.

> .Example of a `interfaceUp` event clearing a `interfaceDown` alarm

Reviewer: ```suggestion
.Example of an `interfaceUp` event clearing an `interfaceDown` alarm
```

[source, xml]
----
   <event>
      <uei>uei.opennms.org/nodes/interfaceDown</uei>
      ...
      <alarm-data reduction-key="%uei%:%dpname%:%nodeid%:%interface%"<1>
                  alarm-type="1"
                  auto-clean="false"/>
   </event>

    <event>
      <uei>uei.opennms.org/nodes/interfaceUp</uei>
      ...
      <alarm-data reduction-key="%uei%:%dpname%:%nodeid%:%interface%"
                  alarm-type="2"
                  clear-key="uei.opennms.org/nodes/interfaceDown:%dpname%:%nodeid%:%interface%"<2>
                  auto-clean="false"/>
   </event>
----

> <1> The `interfaceDown` event sets a `reduction-key` that include enough info to identify a specific interface on a specific node.

Reviewer: ```suggestion
<1> The `interfaceDown` event sets a `reduction-key` that includes enough information to identify a specific interface on a specific node.
```
